intbot/core/endpoints/webhooks.py

```diff
@@ -84,6 +84,54 @@ def verify_github_signature(request) -> str:
     expected = "sha256=" + hashed.hexdigest()
 
     if not hmac.compare_digest(expected, signature):
-        raise ValueError("Signature's don't match")
+        raise ValueError("Signatures don't match")
+
+    return signature
+
+
+@csrf_exempt
+def zammad_webhook_endpoint(request):
+    if request.method == "POST":
+        zammad_headers = {
+            k: v for k, v in request.headers.items() if k.startswith("X-Zammad")
+        }
+
+        try:
+            signature = verify_zammad_signature(request)
+        except ValueError as e:
+            return HttpResponseForbidden(e)
```

Comment:
First you extract the Zammad headers, then you check if the request comes from Zammad and if not you you raise an exception, then you pass Zammad headers to create a webhook. 

Shouldn't it be the other way round - first you check if it comes from Zammad, if so then you extract the headers and pass it on? I'd change the order here, because for a second I thought the headers are passed to verify the signature, but they're passed further down.

Reply:
Thanks for noticing that! In v1 it was a bit different, but it's a leftover from previous change. I updated that for github as well now.